src/UI.elm

```diff
@@ -0,0 +1,34 @@
+module UI exposing (showBoard)
+
+import Html exposing (Html, div, text, button)
+import Html.App as Html
+import Board exposing (Board)
+import Array exposing (Array, length, map, initialize, slice, toList)
+
+showBoard : Board -> Html a
+showBoard board =
+  div [] (showRows board)
+
+
+showRows : Board -> List (Html a)
+showRows board =
+  (create2DBoard board)
+    |> List.map (\line -> div [] (showCells line)) 
+
+
+showCells line =
+  line
+    |> List.map (\cell -> button [] [(text cell)])
+
+
+create2DBoard : Board -> List (List String)
+create2DBoard board =
+  let
+      width = round (sqrt (toFloat (length board)))
```

Comment:
`length board` knows `type alias Board = Array String` on `src/Board.elm:5`.

If you refactor the internals of `Board` as when you did by changing it from `List String` to `Array String`, this might break...

Is there a way to remove that knowledge?